packages/compiler-ssr/src/transforms/ssrTransformTransitionGroup.ts

```diff
@@ -58,14 +60,17 @@ export function ssrProcessTransitionGroup(
 ) {
   const entry = wipMap.get(node)
   if (entry) {
-    const { tag, propsExp } = entry
+    const { tag, propsExp, scopeId } = entry
     if (tag.type === NodeTypes.DIRECTIVE) {
       // dynamic :tag
       context.pushStringPart(`<`)
       context.pushStringPart(tag.exp!)
       if (propsExp) {
         context.pushStringPart(propsExp)
       }
+      if (scopeId) {
+        context.pushStringPart(` ${scopeId}`)
```

Comment:
Changes made: we need to ensure a space before the scopeId since `propsExp` can either be non-existent or render an empty string. If there is no space, the scopeId will be connected to the open tag name and cause error.